packages/block-library/src/navigation-menu/menu-item-inserter.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * WordPress dependencies
+ */
+import { compose } from '@wordpress/compose';
+import { withDispatch } from '@wordpress/data';
+import {
+	Dropdown,
+	IconButton,
+	MenuItem,
+	NavigableMenu,
+	TextControl,
+} from '@wordpress/components';
+import { Component } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { createBlock } from '@wordpress/blocks';
+import { isURL } from '@wordpress/url';
+
+class MenuItemInserter extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			currentSearchInput: '',
+			isUrlInput: false,
+		};
+		this.onChangeSearchInput = this.onChangeSearchInput.bind( this );
+	}
+
+	onChangeSearchInput( value ) {
+		this.setState( {
```

Comment:
You can experiment with `useState` and `useCallback` React hooks :)

Optional, I read https://overreacted.io/a-complete-guide-to-useeffect/ today from @gaearon and got inspired 😃